File: java/01-commands/tests/sourcecode/src/main/java/com/training/app/MultiplesOf3or5/MultiplesOf3or5.java
package com.training.app.MultiplesOf3or5;

public class MultiplesOf3or5 {
    private int number;

    public MultiplesOf3or5(int number) {
        this.number = number;
    }

    public int sum() {
        if (number < 0) return 0;

        int sum = 0;

        while (number > 0) {
            number--;
            if (number % 3 == 0 || number % 5 == 0) {
                sum += number;
            }
        }

        return sum;
    }
}

File: nodejs/01-commands/tests/sourcecode/MultiplesOf3or5/MultiplesOf3or5.js
class MultiplesOf3or5 {
    constructor(number) {
        this.number = number;
    }

    sum() {
        let sum = 0;
        for (let i = 1; i < this.number; i++) {
            if (i % 3 === 0 || i % 5 === 0) {
                sum += i;
            }
        }
        return sum;
    }
}

module.exports = MultiplesOf3or5;